earlystopping-medianstop/rockcraft.yaml

```diff
@@ -1,65 +1,68 @@
+# Based on https://github.com/kubeflow/katib/tree/master/cmd/earlystopping/medianstop/v1beta1
 name: earlystopping-medianstop
 base: ubuntu:22.04
-version: '1-beta1'
+version: 'v0.15.0_22.04_1'
 summary: Early stopping algorithm for katib.
 description: |
     The median stopping rule stops a pending trial X at step S if the trial’s best 
     objective value by step S is worse than the median value of the running averages 
     of all completed trials' objectives reported up to step S.
 license: GPL-3.0
-platforms:
-    amd64: {}
-    # original dockerfile supports also ppc64le and arm64;
-    # at the moment we only support amd64.
-
+run-user: _daemon_
 services:
-  katib-earlystopping-medianstop:
-    command: /usr/bin/katib-earlystopping-medianstop
+  earlystopping-medianstop:
     override: replace
+    summary: "earlystopping-medianstop service"
     startup: enabled
+    command: "python3 /opt/katib/cmd/earlystopping/medianstop/v1beta1/main.py"
+    environment:
+      PYTHONPATH: "/usr/local/lib/python3.10/dist-packages/:/opt/katib:/opt/katib/pkg/apis/manager/v1beta1/python"
+platforms:
+    amd64:
 
 parts:
-    medianstop:
-      plugin: dump
-      source: https://github.com/kubeflow/katib.git
-      source-depth: 1
-      source-type: git
-      source-subdir: cmd/earlystopping/medianstop/v1beta1
-      organize:
-        main.py: opt/katib/cmd/earlystopping/medianstop/v1beta1/main.py
-        requirements.txt: opt/katib/cmd/earlystopping/medianstop/v1beta1/requirements.txt
-      stage:
-        - opt/katib/cmd/earlystopping/medianstop/v1beta1/main.py
-        - opt/katib/cmd/earlystopping/medianstop/v1beta1/requirements.txt
-
-    medianstop-dependencies:
+    # install requirements and other dependencies
+    requirements-and-deps:
       plugin: python
       source: https://github.com/kubeflow/katib.git
+      source-tag: "v0.15.0"
       source-depth: 1
-      source-subdir: cmd/earlystopping/medianstop/v1beta1
       source-type: git
-      source-tag: master
+      source-subdir: cmd/earlystopping/medianstop/v1beta1
+      build-packages:
+        - python3-pip
+        - python3.10
+        - python3.10-venv
+        - python3-dev
+      stage-packages:
+        - python3.10
+        - python3.10-venv
       python-requirements:
         - requirements.txt
+      override-build: |
+        craftctl default
```

Comment:
why do we need this ?

Reply:
Just a left over from workshop. We thought we needed it.
I will re-move and retest.